FE-feat-createAccountButton 가계부 생성 페이지 수정 및 생성 버튼 기능개발

JunYoung7 opened the pull request:
### 📕 Issue Number

Linked #65 
<br/>

### 📗 작업 내역

> 구현 내용 및 작업 했던 내역

- 가계부 생성 페이지 입력 박스 수정
![image](https://user-images.githubusercontent.com/61405355/101528828-21a0c880-39d3-11eb-8b15-1847e3f709e4.png)
   - 이름 입력 부분을 TextArea atom을 이용하여 가계부 이름을 입력할 수 있도록 수정

<br/>

- TextArea atom 수정
![image](https://user-images.githubusercontent.com/61405355/101529015-53b22a80-39d3-11eb-85bd-c8745d743ec7.png)
   - 배경색, fontSize, fontColor를 조절할 수 있도록 수정
   - 입력한 내용을 가져올 수 있도록 onChange 이벤트 함수 등록

<br/>


- 입력한 내용을 가져올 수 있도록 formBox에서 useState로 값을 가질 수 있도록 설정
   - page단에서 입력하는 내용을 useState로 가지고 props로 넘겨주게 되면, 입력 내용이 계속 변하면서 페이지가 계속 rerendering 되기 때문에 page단에서 버튼 이벤트 함수를 넘겨 파라미터로 값을 가져올 수 있도록 구현

<br/>

### 📘 작업 유형

- 신규 기능 추가
- 리펙토링

<br/>

### 📝 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- store를 따로 구현하지 않고 입력 값을 사용할 수 있도록 구현해보았습니다. rerendering을 하지 않도록 하위 컴포넌트에서 state를 유지하도록 구현하였습니다.


<br/><br/>

JunYoung7 committed:
feat : 소셜 가계부 추가 api 생성


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
refactor : TextArea 성능 확장 및 개선

- value값을 사용하기 위한 value 속성 추가
- 배경 색 조절을 위한 backgroundColor 속성 추가
- 글씨 크기, 색깔을 조절하기 위한 속성들 추가 및 기본값 설정
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
refactor : 가계부 생성 페이지 수정

# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명
- name, description value 관리를 위한 useState
- 기존에 가계부 이름 입력 영역을 p태그에서 textArea로 변경

JunYoung7 committed:
refactor : CreateAccountbookPage 수정

- createButtonClick 이벤트를 상위 컴포넌트에서 관리하기 위해 page에서 선언

JunYoung7 committed:
refactor : CreateAccountbookFormBox 수정

# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

- buttonEvent의 파라미터에 넘겨줄 data를 하위 컴포넌트에서 전달해줄 수 있도록 구조 구현

JunYoung7 committed:
refactor : CreateAccountbookFormBox stories 수정 


# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

do02reen24 commented on client/src/components/organisms/CreateAccountbookFormBox/CreateAccountbookFormBox.tsx:
```diff
@@ -9,11 +9,24 @@ import colorUtils from '@utils/color';
   const fontColor = colorUtils.getFontColor(backgroundColor);
   const buttonColor = fontColor === 'white' ? 'black' : 'white';
+
+  const [name, setName] = useState('이름을 입력해주세요');
```
이름의 초기값은 `''`로 지정해주시고 placeholder 값으로 넘겨주는게 좋을 것 같아요! 지금처럼 하면 사용자가 값을 입력하지 않으면 가계부 입력이 이름을 입력해주세요!로 들어가는 문제가 있을 것 같습니다 ㅜㅜ

leejiwoo2002 replied:
저도 여기에 한표 👍

JunYoung7 replied:
placeholder로 바꾸고 속성값도 반영되도록 수정 완료 했습니다. 피드백 감사합니다 도연님 지우님 : )

do02reen24 reviewed (commented):
수고하셨습니다😁👍

Zigje9 commented on client/src/components/organisms/CreateAccountbookFormBox/CreateAccountbookFormBox.stories.tsx:
```diff
@@ -7,8 +7,12 @@ export default {
   component: CreateAccountbookFormBox,
 };
 
+const testFunc = () => {
+  console.log('test');
+};
```
이부분은 삭제하는게 좋을 것 같습니다.

JunYoung7 replied:
넵 data형식으로 넘겨받도록 수정하였습니다 : )

leejiwoo2002 commented on client/src/views/CreateAccountbookPage.tsx:
```diff
@@ -6,15 +6,30 @@ import CreateAccountbookFormBox from '@organisms/CreateAccountbookFormBox';
 import TopNavBar from '@organisms/TopNavBar';
 import CreateAccountbookSetting from '@organisms/CreateAccountbookSetting';
 import colorUtils from '@utils/color';
+/* import * as Axios from '@utils/axios';
+import * as API from '@utils/api'; */
```
주석 부분 날려주셔도 될 것 같습니다!

JunYoung7 replied:
넵 axios 구현과 함께 주석 부분 제거하겠습니다 !

Zigje9 commented on client/src/views/CreateAccountbookPage.tsx:
```diff
@@ -6,15 +6,30 @@ import CreateAccountbookFormBox from '@organisms/CreateAccountbookFormBox';
+import * as API from '@utils/api'; */
 
 const CreateAccountbookPage: React.FC = () => {
+  const createButtonClick = async (data: any) => {
+    /* const data = {
+      name,
+      description,
+      usersList,
+      color,
+    };
+    const result = await Axios.postAxios(API.POST_CREATE_SOCIAL, data); */
+    console.log(data);
+  };
```
파라미터로 받을 수 있다는게 신기하네요! 👍

JunYoung7 replied:
나중에 사용해보셔도 괜찮을 것 같습니다 :)

Zigje9 reviewed (approved):
새벽까지 고생하셨습니다.

JunYoung7 committed:
refactor : TextArea 수정

- placeholder 사용을 위해 placeholder 속성, style 추가

JunYoung7 committed:
refactor : createAccountbookFormBox 수정

- buttonEvent 속성을 함수형태로 변경
- onClick 이벤트를 함수 형태로 사용할 수 있도록 명시적으로 표시
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 reviewed (commented):
피드백 반영 완료

JunYoung7 committed:
Merge branch 'develop' into FULL-feat-createAccountButton
